### PR TITLE
[cxx-interop] Import move only type support by default.

### PR DESCRIPTION
**Pull request by zoecarver**
Just enables importing them by default and adds a new test for `std::unique_ptr`.

Refs https://github.com/apple/swift/pull/65878 https://github.com/apple/swift/pull/65577

### CODE REVIEW

**Comment by zoecarver** [comment]
@swift-ci please smoke test

File: test/Interop/Cxx/class/type-classification-module-interface.swift [hunk 1]
```diff
@@ -24,6 +18,14 @@
...
 // CHECK-NOT: StructWithSubobjectDeletedDestructor
 
+// CHECK: struct StructWithMoveConstructor
+
+// CHECK: struct StructWithInheritedMoveConstructor
+
+// CHECK: struct StructWithSubobjectMoveConstructor
+
+// CHECK: struct StructNonCopyableTriviallyMovable
+
```
Comment (egorzhdan) [inline test/Interop/Cxx/class/type-classification-module-interface.swift:28]: Could you please bring back the checks for `StructWithPointerNonCopyableTriviallyMovable`
and `StructWithPointerNonCopyableTriviallyMovableField`?

File: test/Interop/Cxx/stdlib/std-unique-ptr.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-experimental-cxx-interop)
+//
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import StdUniquePtr
+#if os(Linux)
+import CxxStdlib
+// FIXME: import CxxStdlib.string once libstdc++ is split into submodules.
```
Comment (egorzhdan) [inline test/Interop/Cxx/stdlib/std-unique-ptr.swift:9]: We're not going to split libstdc++ into submodules, let's just import CxxStdlib

**Comment by hyp** [comment]
Let's close this in favor of an updated PR here: https://github.com/apple/swift/pull/69790 . I rebased your initial commits there.

**Pull request closed without merging**